src/main/java/com/shaft/gui/element/ElementActions.java

```diff
@@ -33,2118 +33,2132 @@
+
+	public ElementActions(WebDriver driver) {
+		setLastUsedDriver(driver);
+	}
+
+	public static void setLastUsedDriver(WebDriver driver) {
+		lastUsedDriver = driver;
+	}
+
+	public static String getAiReferenceFileName() {
+		return AI_REFERENCE_FILE_NAME;
+	}
+
+	public static By getAiGeneratedElementLocator() {
+		return aiGeneratedElementLocator;
+	}
+
+	private static void passAction(WebDriver driver) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(driver, null, actionName, null, null);
+	}
+
+	static void passAction(WebDriver driver, By elementLocator) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(driver, elementLocator, actionName, null, null);
+	}
+
+	static void passAction(WebDriver driver, By elementLocator, List<Object> screenshot) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(driver, elementLocator, actionName, null, screenshot);
+	}
+
+	private static void passAction(WebDriver driver, String testData) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(driver, null, actionName, testData, null);
+	}
+
+	static void passAction(WebDriver driver, By elementLocator, String testData) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(driver, elementLocator, actionName, testData, null);
+	}
+
+	static void passAction(WebDriver driver, By elementLocator, String testData, List<Object> screenshot) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(driver, elementLocator, actionName, testData, screenshot);
+	}
+
+	static void passAction(Screen screen, App applicationWindow, Pattern element, String testData) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		passAction(null, null, actionName, testData,
+				SikuliActions.prepareElementScreenshotAttachment(screen, applicationWindow, element, actionName, true));
+	}
+
+	private static void passAction(WebDriver driver, By elementLocator, String actionName, String testData,
+			List<Object> screenshot) {
+		reportActionResult(driver, actionName, testData, elementLocator, screenshot, true);
+	}
+
+	static void failAction(WebDriver driver, By elementLocator, Exception... rootCauseException) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		failAction(driver, actionName, null, elementLocator, null, rootCauseException);
+	}
+
+	static void failAction(WebDriver driver, String testData, By elementLocator, Exception... rootCauseException) {
+		String actionName = Thread.currentThread().getStackTrace()[2].getMethodName();
+		failAction(driver, actionName, testData, elementLocator, null, rootCauseException);
+	}
+
+	static void failAction(Screen screen, App applicationWindow, Pattern element, String testData,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5692539)